sunpy/net/dataretriever/sources/goes.py

```diff
@@ -125,7 +125,7 @@ def _get_url_for_timerange(self, timerange, **kwargs):
         if timerange.end < parse_time("1999/01/15"):
             goes_file = "%Y/go{satellitenumber:02d}%y%m%d.fits"
         elif timerange.start < parse_time("1999/01/15") and timerange.end >= parse_time("1999/01/15"):
-            return self._get_overlap_urls(timerange)
+            return self._get_overlap_urls(timerange, **kwargs)
```

Comment:
What are the kwargs for?

Reply:
Kwargs had satellite number as argument. So as overlap URL method will call `get_url_for_timerange` two times, (when timerange overlaps a certain date) , the information about which satellite number it have to search for is not lost.